polyfill/lib/casts.mjs

```diff
@@ -0,0 +1,140 @@
+import { ES } from './ecmascript.mjs';
+
+export function absolute(arg) {
+  const Absolute = ES.GetIntrinsic('%Temporal.Absolute%');
+  if (arg instanceof Absolute) return arg;
+  if ('bigint' === typeof arg) return new Absolute(arg);
+  if ('string' === typeof arg) {
+    try {
+      return Absolute.fromString(arg);
+    } catch (ex) {}
+  }
+  if (Number.isFinite(+arg)) return Absolute.fromEpochMilliseconds(+arg);
+  throw RangeError(`invalid absolute value: ${arg}`);
+}
+
+export function datetime(arg, add) {
+  const DateTime = ES.GetIntrinsic('%Temporal.DateTime%');
+  if (arg instanceof DateTime) return arg;
+  if ('string' === typeof arg) {
+    try {
+      return DateTime.fromString(arg);
+    } catch (ex) {}
+  }
+  if ('bigint' === typeof arg || 'number' === typeof arg || Number.isFinite(+arg)) return absolute(arg).inZone('UTC');
+  if ('object' === typeof arg) {
+    const { year, month, day } = time('object' === typeof add ? add : arg);
+    const { hour, minute, second, millisecond, microsecond, nanosecond } = date(arg);
+    return new DateTime(year, month, day, hour, minute, second, millisecond, microsecond, nanosecond);
+  }
+  throw new RangeError(`invalid datetime ${arg}`);
+}
+
+export function date(arg) {
+  const Date = ES.GetIntrinsic('%Temporal.Date%');
+  if (arg instanceof Date) return arg;
+  if ('string' === typeof arg) {
+    try {
+      return Date.fromString(arg);
+    } catch (ex) {}
+  }
+  if ('bigint' === typeof arg || 'number' === typeof arg || Number.isFinite(+arg))
+    return absolute(arg)
+      .getDate()
+      .inZone('UTC')
+      .getDate();
```

Comment:
Presumably all of these calls are to the original underly algorithm, not the current method, right?

Reply:
Yes, this generally holds for the entire polyfill. If there is a method function being used that has an underlying algorithm, it's just a code saving measure. I didn't do this when starting this version of the polyfill, but did get back to that during the compatibility & performance drive.